src/Worker/Grpc/GrpcDurableTaskWorker.Processor.cs

```diff
@@ -374,6 +377,31 @@ async Task OnRunOrchestratorAsync(
                     entityConversionState,
                     cancellationToken);
 
+                bool filterPassed = true;
+                if (this.orchestrationFilter != null)
+                {
+                    filterPassed = await this.orchestrationFilter.IsOrchestrationValidAsync(
+                        new OrchestrationInfo
+                        {
+                            Name = runtimeState.Name,
+                            Tags = new Dictionary<string, string>(runtimeState.Tags ?? ImmutableDictionary<string, string>.Empty),
```

Comment:
We shouldn't allocate a new dictionary object if `runtimeState.Tags` is null. Better to just leave the `Tags` property unassigned.

Reply:
Updated.